compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -438,6 +439,164 @@ LogicalResult GatherOp::generateScalarImplementation(OpBuilder &b, Location loc,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// MapScatterOp
+//===----------------------------------------------------------------------===//
+
+SmallVector<utils::IteratorType> MapScatterOp::getLoopIteratorTypes() {
+  SmallVector<utils::IteratorType> iteratorTypes(getInputRank(),
+                                                 utils::IteratorType::parallel);
+  return iteratorTypes;
+}
+
+SmallVector<Range> MapScatterOp::getIterationDomain(OpBuilder &builder) {
+  Location loc = getLoc();
+  OpFoldResult zero = builder.getIndexAttr(0);
+  OpFoldResult one = builder.getIndexAttr(1);
+  SmallVector<Range> ranges;
+  for (auto dim : llvm::seq<int64_t>(0, getInputRank())) {
+    OpFoldResult ub = getDim(builder, loc, getInput(), dim);
+    ranges.push_back(Range{zero, ub, one});
+  }
+  return ranges;
+}
+
+/// Generate the tiled implemenation of the map_scatter op, which scatters a
+/// slice of the input (taken from the input with the `offsets` and `sizes`)
+/// to the full output tensor, and inserts an addition of the `offsets` at the
+/// start of the transformation_region body.
+FailureOr<TilingResult>
+MapScatterOp::getTiledImplementation(OpBuilder &builder,
+                                     ArrayRef<OpFoldResult> offsets,
+                                     ArrayRef<OpFoldResult> sizes) {
+  assert(offsets.size() >= 1 && sizes.size() >= 1);
+  Location loc = getLoc();
+  SmallVector<Operation *> slices;
+
+  // Slice of the input.
+  auto oneAttr = builder.getI64IntegerAttr(1);
+  SmallVector<OpFoldResult> inputStrides(getInputRank(), oneAttr);
+  Operation *inputSlice =
+      getSlice(builder, loc, getInput(), offsets, sizes, inputStrides);
+  if (!inputSlice) {
+    return emitOpError("failed to get input slice");
+  }
+  Value tiledInput = inputSlice->getResult(0);
+  slices.push_back(inputSlice);
+
+  Operation *tiledOp = mlir::clone(builder, getOperation(), getResultTypes(),
```

Comment:
Interesting, I didnt realize you never planned to tile the output. So in some sense it ends up with some sort of slicing semantics. I have to think about how this affects things.